<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from urdf/irb4600_60_205.xacro      | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="abb_irb4600_60_205">
  <!-- link list -->
  
  <link name="base_link">
    <collision name="collision">
      <geometry>
        <mesh filename="package://abb_irb4600_support/meshes/irb4600/collision/base_link.stl"/>
      </geometry>
    </collision>
    <visual name="visual">
      <geometry>
        <mesh filename="package://abb_irb4600_support/meshes/irb4600/visual/base_link.dae"/>
      </geometry>
      <material name="yellow">
        <color rgba="1 1 0 1"/>
      </material>
    </visual>
    <inertial>
      <mass value="1.0" />
      <inertia ixx="0.2083" ixy="0.0" ixz="0.0" iyy="0.2083" iyz="0.0" izz="0.3333"/>
    </inertial>
  </link>
  <link name="link_1">
    <collision name="collision">
      <geometry>
        <mesh filename="package://abb_irb4600_support/meshes/irb4600/collision/link_1.stl"/>
      </geometry>
    </collision>
    <visual name="visual">
      <geometry>
        <mesh filename="package://abb_irb4600_support/meshes/irb4600/visual/link_1.dae"/>
      </geometry>
      <material name="yellow"/>
    </visual>
    <inertial>
      <mass value="1.0" />
      <inertia ixx="0.2083" ixy="0.0" ixz="0.0" iyy="0.2083" iyz="0.0" izz="0.3333"/>
    </inertial>
  </link>
  <link name="link_2">
    <collision name="collision">
      <geometry>
        <mesh filename="package://abb_irb4600_support/meshes/irb4600_60_205/collision/link_2.stl"/>
      </geometry>
    </collision>
    <visual name="visual">
      <geometry>
        <mesh filename="package://abb_irb4600_support/meshes/irb4600_60_205/visual/link_2.dae"/>
      </geometry>
      <material name="yellow"/>
    </visual>
    <inertial>
      <mass value="1.0" />
      <inertia ixx="0.2083" ixy="0.0" ixz="0.0" iyy="0.2083" iyz="0.0" izz="0.3333"/>
    </inertial>
  </link>
  <link name="link_3">
    <collision name="collision">
      <geometry>
        <mesh filename="package://abb_irb4600_support/meshes/irb4600_60_205/collision/link_3.stl"/>
      </geometry>
    </collision>
    <visual name="visual">
      <geometry>
        <mesh filename="package://abb_irb4600_support/meshes/irb4600_60_205/visual/link_3.dae"/>
      </geometry>
      <material name="yellow"/>
    </visual>
    <inertial>
      <mass value="1.0" />
      <inertia ixx="0.2083" ixy="0.0" ixz="0.0" iyy="0.2083" iyz="0.0" izz="0.3333"/>
    </inertial>
  </link>
  <link name="link_4">
    <collision name="collision">
      <geometry>
        <mesh filename="package://abb_irb4600_support/meshes/irb4600_60_205/collision/link_4.stl"/>
      </geometry>
    </collision>
    <visual name="visual">
      <geometry>
        <mesh filename="package://abb_irb4600_support/meshes/irb4600_60_205/visual/link_4.dae"/>
      </geometry>
      <material name="yellow"/>
    </visual>
    <inertial>
      <mass value="1.0" />
      <inertia ixx="0.2083" ixy="0.0" ixz="0.0" iyy="0.2083" iyz="0.0" izz="0.3333"/>
    </inertial>
  </link>
  <link name="link_5">
    <collision name="collision">
      <geometry>
        <mesh filename="package://abb_irb4600_support/meshes/irb4600_60_205/collision/link_5.stl"/>
      </geometry>
    </collision>
    <visual name="visual">
      <geometry>
        <mesh filename="package://abb_irb4600_support/meshes/irb4600_60_205/visual/link_5.dae"/>
      </geometry>
      <material name="yellow"/>
    </visual>
    <inertial>
      <mass value="1.0" />
      <inertia ixx="0.2083" ixy="0.0" ixz="0.0" iyy="0.2083" iyz="0.0" izz="0.3333"/>
    </inertial>
  </link>
  <link name="link_6">
    <collision name="collision">
      <geometry>
        <mesh filename="package://abb_irb4600_support/meshes/irb4600_60_205/collision/link_6.stl"/>
      </geometry>
    </collision>
    <visual name="visual">
      <geometry>
        <mesh filename="package://abb_irb4600_support/meshes/irb4600_60_205/visual/link_6.dae"/>
      </geometry>
      <material name="yellow"/>
    </visual>
    <inertial>
      <mass value="1.0" />
      <inertia ixx="0.2083" ixy="0.0" ixz="0.0" iyy="0.2083" iyz="0.0" izz="0.3333"/>
    </inertial>
  </link>
  <!-- end of link list -->
  <!-- joint list -->
  <joint name="joint_1" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0.495"/>
    <axis xyz="0 0 1"/>
    <parent link="base_link"/>
    <child link="link_1"/>
    <limit effort="0" lower="-3.141" upper="3.141" velocity="3.054"/>
  </joint>
  <joint name="joint_2" type="revolute">
    <origin rpy="0 0 0" xyz="0.175 0 0"/>
    <axis xyz="0 1 0"/>
    <parent link="link_1"/>
    <child link="link_2"/>
    <limit effort="0" lower="-1.570" upper="2.617" velocity="3.054"/>
  </joint>
  <joint name="joint_3" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0.9"/>
    <axis xyz="0 1 0"/>
    <parent link="link_2"/>
    <child link="link_3"/>
    <limit effort="0" lower="-3.141" upper="1.308" velocity="3.054"/>
  </joint>
  <joint name="joint_4" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0.175"/>
    <axis xyz="1 0 0"/>
    <parent link="link_3"/>
    <child link="link_4"/>
    <limit effort="0" lower="-6.981" upper="6.981" velocity="4.363"/>
  </joint>
  <joint name="joint_5" type="revolute">
    <origin rpy="0 0 0" xyz="0.960 0 0 "/>
    <axis xyz="0 1 0"/>
    <parent link="link_4"/>
    <child link="link_5"/>
    <limit effort="0" lower="-2.181" upper="2.094" velocity="4.363"/>
  </joint>
  <joint name="joint_6" type="revolute">
    <origin rpy="0 0 0" xyz="0.135 0 0 "/>
    <axis xyz="1 0 0"/>
    <parent link="link_5"/>
    <child link="link_6"/>
    <limit effort="0" lower="-6.981" upper="6.981" velocity="6.283" />
  </joint>
  <!-- end of joint list -->
  <!-- ROS-Industrial 'base' frame: base_link to ABB World Coordinates transform -->
  <link name="base"/>
  <joint name="base_link-base" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="base_link"/>
    <child link="base"/>
  </joint>
  <!-- ROS-Industrial 'flange' frame: attachment point for EEF models -->
  <link name="flange"/>
  <joint name="joint_6-flange" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="link_6"/>
    <child link="flange"/>
  </joint>
  <!-- ROS-Industrial 'tool0' frame: all-zeros tool frame -->
  <link name="tool0">
    <collision>
      <geometry>
        <box size="0.05 0.05 0.05"/>
      </geometry>
    </collision>
    <visual>
      <geometry>
        <box size="0.05 0.05 0.05"/>
      </geometry>
    </visual>
    <inertial>
      <mass value="1.0" />
      <inertia ixx="0.2083" ixy="0.0" ixz="0.0" iyy="0.2083" iyz="0.0" izz="0.3333"/>
    </inertial>
  </link>
  <joint name="link_6-tool0" type="fixed">
    <origin rpy="0 1.5707963267948966 0" xyz="0 0 0"/>
    <parent link="flange"/>
    <child link="tool0"/>
  </joint>

  <!-- <ros2_control name="GazeboSimSystem" type="system">
    <hardware>
      <plugin>gz_ros2_control/GazeboSimSystem</plugin>
    </hardware> -->
  <ros2_control name="IgnitionSystem" type="system">
    <hardware>
      <plugin>ign_ros2_control/IgnitionSystem</plugin>
    </hardware>
    <!-- Revolute joints -->
    <joint name="joint_1">
      <command_interface name="position">
        <param name="min">-15</param>
        <param name="max">15</param>
      </command_interface>
      <state_interface name="position">
        <param name="initial_value">0.0</param> 
      </state_interface>
      <state_interface name="velocity"/>
    </joint>
    <joint name="joint_2">
      <command_interface name="position" />
      <state_interface name="position">
        <param name="initial_value">0.0</param> 
      </state_interface>
      <state_interface name="velocity"/>
    </joint>
    <joint name="joint_3">
      <command_interface name="position" />
      <state_interface name="position">
        <param name="initial_value">0.0</param> 
      </state_interface>
      <state_interface name="velocity"/>
    </joint>
    <joint name="joint_4">
      <command_interface name="position" />
      <state_interface name="position">
        <param name="initial_value">0.0</param> 
      </state_interface>
      <state_interface name="velocity"/>
    </joint>
    <joint name="joint_5">
      <command_interface name="position" />
      <state_interface name="position">
        <param name="initial_value">0.0</param> 
      </state_interface>
      <state_interface name="velocity"/>
    </joint>
    <joint name="joint_6">
      <command_interface name="position" />
      <state_interface name="position">
        <param name="initial_value">0.0</param> 
      </state_interface>
      <state_interface name="velocity"/>
    </joint>
  </ros2_control>

  <gazebo>  
    <plugin filename="ign_ros2_control-system" name="ign_ros2_control::IgnitionROS2ControlPlugin">
      <parameters>/home/yangjingyun/robots/ros2_simulation/src/warehouse_manage/config/abb_irb4600_60_205.yaml</parameters>
    </plugin>
  </gazebo>
  <gazebo reference="base_link">
    <turnGravityOff>true</turnGravityOff>
  </gazebo>
  <gazebo reference="link_1">
    <turnGravityOff>true</turnGravityOff>
  </gazebo>
  <gazebo reference="link_2">
    <turnGravityOff>true</turnGravityOff>
  </gazebo>
  <gazebo reference="link_3">
    <turnGravityOff>true</turnGravityOff>
  </gazebo>
  <gazebo reference="link_4">
    <turnGravityOff>true</turnGravityOff>
  </gazebo>
  <gazebo reference="link_5">
    <turnGravityOff>true</turnGravityOff>
  </gazebo>
  <gazebo reference="link_6">
    <turnGravityOff>true</turnGravityOff>
  </gazebo>
  <gazebo reference="flange">
    <turnGravityOff>true</turnGravityOff>
  </gazebo>
  <gazebo reference="tool0">
    <turnGravityOff>true</turnGravityOff>
  </gazebo>
</robot>
